model_compression_toolkit/common/substitutions/layer_fusing.py

```diff
@@ -0,0 +1,101 @@
+# Copyright 2022 Sony Semiconductors Israel, Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+import copy
+from typing import Any, List, Tuple
+from model_compression_toolkit.common.graph.base_graph import Graph
+from model_compression_toolkit.common.graph.base_node import BaseNode
+from model_compression_toolkit.common.target_platform.targetplatform2framework import TargetPlatformCapabilities
+from model_compression_toolkit.common.target_platform.targetplatform2framework.layer_filter_params import LayerFilterParams
+
+def filter_fusing_patterns(fusing_patterns: List[List[Any]], node: BaseNode, idx: int = 0):
+    """
+    Update relevant fusing patterns object if layer number 'idx' inside the fusion matches the node
+    Args:
+        fusing_patterns: supported fusings
+        node: node to decide if it can be a part of fusion
+        idx: index of layer in the fusion
+    Returns:
+        fusing_patterns after filtering non-relevant fusions
+    """
+    valid_fusing_patterns = []
+    for i,fusing_pattern in enumerate(fusing_patterns):
+        if idx < len(fusing_pattern):
+            if (type(fusing_pattern[idx]) == LayerFilterParams and fusing_pattern[idx].match(node)) or fusing_pattern[idx] == node.type:
+                valid_fusing_patterns.append(fusing_pattern)
+
+    # Return only valid patterns for this node
+    return valid_fusing_patterns
+
+
+def mark_fusing_activation(nodes):
+    for node in nodes:
+        for qc in node.candidates_quantization_cfg:
+            qc.activation_quantization_cfg.enable_activation_quantization = False
+
+
+class LayerFusing(object):
```

Comment:
This is not the right place for this file, please create a fusing package under common.

Reply:
Done